src/test/skiplist_tests.cpp

```diff
@@ -20,13 +20,13 @@ BOOST_AUTO_TEST_CASE(skiplist_test)
 
     for (int i=0; i<SKIPLIST_LENGTH; i++) {
         vIndex[i].nHeight = i;
-        vIndex[i].pprev = (i == 0) ? NULL : &vIndex[i - 1];
+        vIndex[i].pprev = (i == 0) ? NULL : (vIndex.data() + i - 1);
         vIndex[i].BuildSkip();
     }
 
     for (int i=0; i<SKIPLIST_LENGTH; i++) {
         if (i > 0) {
-            BOOST_CHECK(vIndex[i].pskip == &vIndex[vIndex[i].pskip->nHeight]);
+            BOOST_CHECK(vIndex[i].pskip == vIndex.data() + vIndex[i].pskip->nHeight);
```

Comment:
The `.data()` usage here however,  is not really inline with #9804.
#9804 replaced usage of `&v[0]` and `&v[N]` for specifying a slice/range.
In this change above,  you are specifically referring to the memory location of 1 element,  which isn't an rvalue... probably unnecessary to use `.data()`.